app/services/config_data_sources/json_project_data_source.rb

```diff
@@ -226,13 +226,16 @@ def update_project!(project: nil)
         projects = self.projects
         projects[project_index] = project
         self.projects = projects
+
+        return true
```

Comment:
This is weird to me. I get that it can help with testing, but I think the better way to test that this method does as it's supposed to is by ensuring the `project` you updated is actually up-to-date by testing the resulting `update` is better than adding a return statement here. Since we're dealing with state here, I think explicitly checking that our state updated exactly how we expected is 🔑

Reply:
Yeah, I think removing this stuff is the best course of action. Returning `true` isn't really helpful. It is really just a signal that this method exited without raising an exception, and it doesn't provide useful info since you're not going to check for it during runtime.

If we do this in other places, we should probably fix that at some point, would you mind opening an issue that points this out?